src/javascript/app/pages/user/account/authenticate.js

```diff
@@ -221,45 +216,97 @@ const Authenticate = (() => {
             return Promise.all(promises);
         };
 
+        // Save file info to be used for validation and populating the file info
+        const fileTracker = ($ele, track=true) => {
+            const [, id, pos]       = ($ele.attr('id').match(/([a-z\d]+)_(\d)/) || []);
+            if (track) {
+                file_checks[id]            = file_checks[id] || {};
+                file_checks[id].files      = file_checks[id].files || [];
+                file_checks[id].files[pos] = true;
+                file_checks[id].type      = $ele.attr('accept');
+                file_checks[id].label     = $ele.attr('data-name');
+            } else {
+                file_checks[id].files[pos] = false;
+            }
+        };
+
         // Validate user input
         const validate = (file) => {
-            const required_docs = ['passport', 'proofid', 'driverslicense'];
-            const doc_name = {
-                passport      : localize('Passport'),
-                proofid       : localize('Identity card'),
-                driverslicense: localize('Driving licence'),
-            };
+            // Add error messages here. Error messages are mapped by index to checks.
+            const error_messages = [
+                localize('Invalid document format: "[_1]" for [_2]', [file.documentFormat, file_checks[file.id].label]),
+                localize('File ([_1]) size exceeds the permitted limit. Maximum allowed file size: 3MB', [file_checks[file.id].label]),
+                localize('ID number is required for [_1].', [file_checks[file.id].label]),
+                localize('Only letters, numbers, space, underscore, and hyphen are allowed for ID number ([_1]).', [file_checks[file.id].label]),
+                localize('Expiry date is required for [_1].', [file_checks[file.id].label]),
+                localize('Front and reverse side photos of [_1] are required.', [file_checks[file.id].label]),
+                // Japan Error message
+                localize('My number card is required.'),
+                localize('Please select at least one file from section 1 or two different file types from section 2.'),
+                localize('One of [_1] or [_2] is required.', [localize('Medical insurance card'), localize('Pension book')]),
+                localize('One of [_1], [_2] or [_3] is required.', [localize('Resident records'),
```

Comment:
Missing the first string from `generate-static-data.js`. (same for the above line)